src/components/select/multi-select/multi-select.component.js

```diff
@@ -111,6 +116,34 @@ const MultiSelect = React.forwardRef(
       [name, id]
     );
 
+    /* generic value update function which can be used for both controlled and uncontrolled
+     * components, both with and without onChange.
+     * - if onChange is provided, calls that with the updated value, provided it has changed in length
+     * (ie if an option has been selected or deselected)
+     * - if the component is uncontrolled, call setSelectedValue with the provided update function
+     * - if the component is controlled, do not change selectedValue and instead directly call
+     * the update function and pass its result to onChange
+     * - ensure that in every circumstance, onChange and the update function are only called at most
+     * once, in case they have side effects */
```

Comment:
Yes, it is long :( I added it late because I thought it would help to explain everything the function was trying to do - of course it's better if the code is self-documenting but I'm not sure how to do that in this case. I'll take another look!

Reply:
I agree with @grabkowski but understand why you've added this comment @robinzigmond, I think the top two lines would suffice and you could add a short comment above the conditions if you feel it's needed